src/tool_cfgable.h

```diff
@@ -29,6 +29,19 @@
 
 #include "tool_formparse.h"
 
+/* TODO: 
+ *   probably should set up USE_ESNI symbol (in curl_setup.h ?)
+ *   derived from HAVE_OPENSSL_ESNI_H and !OPENSSL_NO_ESNI
+ *   and use it for conditional inclusion of ESNI support code.
+ *   This will allow for adding other (non-OpenSSL) ESNI implementations.
+ *
+ *   For now, we're just using HAVE_OPENSSL_ESNI_H
+ */
+
+#ifdef HAVE_OPENSSL_ESNI_H
+# include <openssl/esni.h>
```

Comment:
I appreciate both this useful advice and that you've taken time to review a code fragment which was removed (in aaf0a3e) so that it would not (as I thought) be exposed in the PR.